src/UserFunctions.java

```diff
@@ -117,12 +135,48 @@ private void loadPuzzle() {
     }
 
     private void guessWord(String word) {
-        //TODO check if the word is in puzzle.validWords
+        if (puzzle == null)
+            return;
+
+        int wasValid = puzzle.guess(word);
+
+        if (wasValid < 0) {
+            System.out.println("You already found this word.");
+
+        } else if (wasValid == 0) {
+            System.out.println("Your guess was not a valid word. Try again!");
+            
+        } else {
+            System.out.println("Good job! Your word was worth " + wasValid + " points.");
+        }
```

Comment:
The gaps here feel a bit weird.

```suggestion
        if (wasValid < 0) {
            System.out.println("You already found this word.");
        } else if (wasValid == 0) {
            System.out.println("Your guess was not a valid word. Try again!");
        } else {
            System.out.println("Good job! Your word was worth " + wasValid + " points.");
        }
```